### PR TITLE
Add script for formatting code using clang-tidy.

### PR DESCRIPTION
**Pull request by smolkaj**
I will actually run the script and check formatting via CI in a separate PR so this one doesn't get too messy.

### CODE REVIEW

File: format.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Formats source files according to Google's style guide. Requires clang-tidy.
+
+# Only files with these extensions will be formatted.
+EXTENSIONS="cc|h|proto"
```
Comment (stefanheule) [inline format.sh:5]: Can we format BUILD files, too?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,3 +13,4 @@ All developers must sign the [P4.org](http://p4.org) CLA and return it to
 
 Any contribution to the C++ core code must respect the coding guidelines.
 We follow the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
+We provide a [script](format.sh) to automatically format all code correctly.
```
Comment (stefanheule) [inline CONTRIBUTING.md:16]: Can we run the tool in CI and reject non-formatted code?

Comment (smolkaj) [reply CONTRIBUTING.md:16]: Yes. Will do in a separate PR (since reformatting will create a lot of noise in the diff).

File: format.sh (hunk 1 above)
Comment (smolkaj) [reply format.sh]: Not with clang-format. But we can with "buildifier". I did so in the last commit.

**Pull request merged**